framework/scripts/generate_test_list.py

```diff
@@ -74,28 +74,32 @@ def main():
 
     # generate the .cpp file
     with open(cpp_file, 'w') as f:
+        f.write('#include <span>\n')
         f.write(f'#include "{h_file_basename}"\n\n')
+        f.write('using namespace std;\n\n')
         decls = [ f'extern struct test _test_{test};' for test in sorted(all_tests.keys()) ]
         f.write('\n'.join(decls))
         f.write('\n\n')
         for name, tests in test_lists.items():
-            f.write(f'static constexpr struct test *{name}_test_list[] = {{')
+            f.write(f'static constexpr struct test * const {name}_test_list_[] = {{')
             test_list = [ f'    &_test_{test}' for test in tests ]
             f.write(',\n'.join(test_list))
             f.write('\n};\n\n')
+            f.write(f'static constinit const span<struct test * const> {name}_test_list({name}_test_list_);\n\n');
         # write selector function
-        f.write('struct test * const *get_test_list(const char *platform, size_t *size) {\n')
+        f.write('const span<struct test * const> *get_test_list(const char *test_list_name) {\n')
```

Comment:
Don't return a pointer to the span. Just return the span.

The list array itself, above, can remain the unchanged. I think you simply return the array in the generated test function code and `std::span` will automatically get its size.

Reply:
I need to know if the function returns nothing, hence returning the nullptr. What would you suggest instead?

Reply:
Did it with optional which is basically the same as a pointer, but in more characters. :)